metricbeat/module/mongodb/dbstats/dbstats.go

```diff
@@ -87,23 +82,23 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 
 		err := db.Run("dbStats", &result)
 		if err != nil {
-			err = errors.Wrapf(err, "Failed to retrieve stats for db %s", dbName)
-			logger.Error(err)
-			continue
+			m.Logger().Error(errors.Wrapf(err, "Failed to retrieve stats for db %s", dbName))
```

Comment:
We lost the `continue` here and are not reporting the error event anymore.

Reply:
Good catch!